kolibri/plugins/management/assets/src/vue/user-page/user-create-modal.vue

```diff
@@ -72,22 +74,46 @@
           full_name: this.full_name,
           facility: this.facility,
         };
-        // using promise to ensure that the user is created before closing
-        this.createUser(newUser, this.role).then(
-          () => {
-            this.full_name = '';
-            this.username = '';
-            this.password = '';
-            this.$refs.modal.closeModal();
-          }).catch((error) => {
-            if (error.status.code === 409) {
-              this.errorMessage = error.entity;
-            } else if (error.status.code === 403) {
-              this.errorMessage = error.entity;
-            } else {
-              this.errorMessage = `Whoops! Something went wrong.`;
-            }
-          });
+        let populated = true;
+
+        if (!newUser.username) {
```

Comment:
These ifs make me sad. But I figured it would be better to have explicit strings.